File: leetcode/217-contains-duplicate/contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> set;

        for (int i = 0; i < n; i++) {
            if (set.count(nums[i])) {
                return true;
            }
            set.insert(nums[i]);
        }
        return false;
    }
};

File: 217-contains-duplicate/contains-duplicate.rs
use std::collections::HashSet;
use std::iter::FromIterator;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let exists: HashSet<i32> = HashSet::from_iter(nums.clone());
        !(&nums.len() == &exists.len())
    }
}